x-pack/plugins/siem/public/components/url_state/normalize_time_range.test.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { normalizeTimeRange } from './normalize_time_range';
+import { URLTimeRange, AbsoluteTimeRange, isAbsoluteTimeRange } from '../../store/inputs/model';
+
+describe('#normalizeTimeRange', () => {
```

Comment:
Seems that we also have the case of normalising relative time range, shall we add some cases for that as well. It would be nice to see the input of relative time range here, but happy to leave it like this as well.

Reply:
Yea, I will add a few tests for this. Thanks for the feedback.

Reply:
Yea, I will add tests for this. Thanks for the feedback.

Reply:
Thanks for adding the cases, Frank. Looks good! I was wondering about what's the difference between `from`/`to` of AbsoluteTimeRange and RelativeTimeRange, seems that it means that they are in the same format.
We use `global.timerange.kind` to decide whether it's an absolute or a relative timerange, but still not sure how `global.timerange.kind` is decided, would like to know what should I do to produce a case for it to be relative in UI?

Reply:
Bear with me, I found it, thanks!
<img width="545" alt="Screenshot 2019-06-18 at 03 17 20" src="https://user-images.githubusercontent.com/6295984/59630343-aa8cca00-9177-11e9-87e9-6469729ca4a4.png">